pkg/validation/spec/swagger.go

```diff
@@ -123,6 +154,18 @@ func (s *SchemaOrBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (s *SchemaOrBool) UnmarshalNextJSON(opts jsonv2.UnmarshalOptions, dec *jsonv2.Decoder) error {
+	if dec.PeekKind() == '{' {
```

Comment:
consider enforcing the correct types, switching on dec.PeekKind() and honoring these cases:
* `{` (set `Allows` to `true`)
* `t` (set `Allows` to `true`)
* `f` (set `Allows` to `false`)

Reply:
added